src/dioptra/restapi/v1/utils.py

```diff
@@ -78,39 +78,31 @@ def build_current_user(user: models.User) -> dict[str, Any]:
 
 
 def build_group(group: models.Group) -> dict[str, Any]:
-    permissions = {
-        x.user.user_id: {
-            "user": build_user_ref(x.user),
-            "group": build_group_ref(x.group),
-            "permissions": {
-                "read": False,
-                "write": False,
-                "share_read": False,
-                "share_write": False,
-                "owner": False,
-                "admin": False,
-            },
-        }
-        for x in group.members + group.managers
-    }
+    members: dict[int, dict[str, Any]] = {}
+
     for member in group.members:
-        permissions[member.user.user_id]["permissions"].update(
-            {
+        members[member.user_id] = {
+            "user": build_user_ref(member.user),
+            "group": build_group_ref(group),
+            "permissions": {
                 "read": member.read,
                 "write": member.write,
                 "share_read": member.share_read,
                 "share_write": member.share_write,
+                "owner": False,
+                "admin": False,
             }
-        )
+        }
+
     for manager in group.managers:
-        permissions[manager.user.user_id]["permissions"].update(
-            {"owner": manager.owner, "admin": manager.admin}
-        )
+        members[manager.user_id]["permissions"]["owner"] = manager.owner
```

Comment:
thought about doing it this way, but couldn't we have a case where a user is a manager but not a member? i.e. they are administering the group but don't need to access its resources?

If we want that case (or if it could happen inadvertently), indexing into "permissions" could fail.